tests/DocBlock/TypeExpressionTest.php

```diff
@@ -1014,6 +1038,16 @@ public static function provideSortTypesCases(): iterable
             '18_446_744_073_709_551_616|-8.2023437675747321e-18_446_744_073_709_551_616',
             '-8.2023437675747321e-18_446_744_073_709_551_616|18_446_744_073_709_551_616',
         ];
+
+        yield 'mixed 2x | and & glue' => [
+            'Foo|Foo2|Baz&Bar',
+            'Bar&Baz|Foo|Foo2',
+        ];
+
+        yield 'mixed | and 2x & glue' => [
+            'Foo|Baz&Baz2&Bar',
+            'Bar&Baz&Baz2|Foo',
+        ];
```

Comment:
I thought we need proper test case for DNF types, but it looks like `conditional with variable` case covers it and works properly. I checked it, because previously you said that `&` has higher priority, but when it comes to DNF types, it's the `|` which is a main glue. [`Foo&(Bar|Baz)` is invalid](https://3v4l.org/7hcfi#v8.2.23), but currently `TypeExpression` parses it and does not throw an exception:

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/93079fbb-7dd4-4b8f-b0f7-3d17ea7548f8">

I am not sure if we should cover this in any way, especially in this PR (probably not). Just sharing what I found.

Reply:
It is a valid phpdoc. And we support it. :)

Reply:
Yeah, it looks like [it's normalized to proper DNF type internally](https://phpstan.org/r/7cb42dc1-3036-47ed-863f-862d26ab7c3c), so we can assume it's valid-ish 👍.